src/kbc_modules/eaa_kbc/rats_tls/mod.rs

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) 2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use foreign_types::{ForeignType, ForeignTypeRef, Opaque};
+use log::*;
+use std::ops::{Deref, DerefMut};
+use std::os::unix::io::RawFd;
+use std::ptr::NonNull;
+
+mod ffi;
+use ffi::*;
+
+pub struct RatsTlsRef(Opaque);
+
+unsafe impl ForeignTypeRef for RatsTlsRef {
+    type CType = rats_tls_handle;
+}
+
+#[derive(Clone)]
+pub struct RatsTls(NonNull<rats_tls_handle>);
+
+unsafe impl Send for RatsTlsRef {}
+unsafe impl Sync for RatsTlsRef {}
+unsafe impl Send for RatsTls {}
+unsafe impl Sync for RatsTls {}
+
+unsafe impl ForeignType for RatsTls {
+    type CType = rats_tls_handle;
+    type Ref = RatsTlsRef;
+
+    unsafe fn from_ptr(ptr: *mut rats_tls_handle) -> RatsTls {
+        RatsTls(NonNull::new_unchecked(ptr))
+    }
+
+    fn as_ptr(&self) -> *mut rats_tls_handle {
+        self.0.as_ptr()
+    }
+
+    fn into_ptr(self) -> *mut rats_tls_handle {
+        let inner = self.as_ptr();
+        ::core::mem::forget(self);
+        inner
+    }
+}
+
+impl Drop for RatsTls {
+    fn drop(&mut self) {
+        unsafe {
+            rats_tls_cleanup(self.as_ptr());
+        }
+    }
+}
+
+impl Deref for RatsTls {
+    type Target = RatsTlsRef;
+
+    fn deref(&self) -> &RatsTlsRef {
+        unsafe { RatsTlsRef::from_ptr(self.as_ptr()) }
+    }
+}
+
+impl DerefMut for RatsTls {
+    fn deref_mut(&mut self) -> &mut RatsTlsRef {
+        unsafe { RatsTlsRef::from_ptr_mut(self.as_ptr()) }
+    }
+}
+
+impl RatsTls {
+    pub fn new(
+        server: bool,
```

Comment:
What does this server bool do? It seems like it should be called `is_server` or something a bit more descriptive.

Reply:
Since rats-tls supports two-way authentication, this bool parameter is used to indicate whether the created rats-tls instance supports server-side authentication. The name `server` should be unambiguous. It's not a big deal.

Reply:
Can we please remove it if we are not planning to use it?